packages/repository/src/__tests__/unit/repositories/define-repository-class.unit.ts

```diff
@@ -29,10 +28,10 @@ import {
 describe('RepositoryClass builder', () => {
   describe('defineCrudRepositoryClass', () => {
     it('should generate CRUD repository class', async () => {
-      const AddressRepository = defineCrudRepositoryClass(
-        Address,
-        DummyCrudRepository,
-      );
+      const AddressRepository = defineCrudRepositoryClass<
+        typeof Address,
+        DummyCrudRepository<Address>
```

Comment:
So TypeScript cannot infer the generic types from the parameters in this case?

Reply:
To make inference happy, I have to do the following:

```ts
function castDummyRepository<M extends Model>() {
  return class extends DummyCrudRepository<M> {};
}

const AddressRepository = defineCrudRepositoryClass(
  Address,
  castDummyRepository<Address>(),
);
```

Reply:
TypeScript does not allow `typeof DummyCrudRepository<M>`.

Reply:
> So TypeScript cannot infer the generic types from the parameters in this case?

Unfortunately not. I think we should report a bug to TypeScript (if it wasn't already reported).

> To make inference happy, I have to do the following (...)

IMO, `castDummyRepository` does not solve the problem of people forgetting to specify the repository type and relying on the compiler to infer it. Once they realize some extra effort is needed, then I think it's best to provide generic type parameters and there is no need to introduce `cast*Repository` functions.